### PR TITLE
manifest: TF-M v2.0.0 update

### PR DESCRIPTION
**Pull request by mswarowsky**
Update TF-M to version 2.0.0 including
MbedTLS to v3.5.0
TF-M-Tests to v2.0.0
PSA-Arch_Tests to v23.06_API1.5_ADAC_EAC

### CODE REVIEW

File: modules/mbedtls/CMakeLists.txt [hunk 1]
```diff
@@ -180,6 +180,7 @@ zephyr_interface_library_named(mbedTLS)
       ${ZEPHYR_CURRENT_MODULE_DIR}/library/ssl_tls13_keys.c
       ${ZEPHYR_CURRENT_MODULE_DIR}/library/ssl_tls13_server.c
       ${ZEPHYR_CURRENT_MODULE_DIR}/library/ssl_tls.c
+      zephyr_time.c
```
Comment (SebastianBoe) [inline modules/mbedtls/CMakeLists.txt:184]: This source file does not look like the others in this list.

It looks more like zephyr_init.c, debug.c, shell.c etc.

Let's move it's CMake code next to one of those three (see above in this file).

File: modules/mbedtls/zephyr_time.c [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mbedtls/platform_time.h>
+#include <zephyr/kernel.h>
+
+mbedtls_ms_time_t mbedtls_ms_time(void)
+{
+	return (mbedtls_ms_time_t)k_uptime_get();
+}
```
Comment (SebastianBoe) [inline modules/mbedtls/zephyr_time.c:13]: Invoking the C compiler for just this little function slows down the build.

Let's move it into zephyr_init.c.

zephyr_init.c already contains Zephyr-specific mbedtls platform implementations like mbedtls_hardware_poll.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: modules/mbedtls/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply modules/mbedtls/CMakeLists.txt:184]: NB: See https://github.com/zephyrproject-rtos/zephyr/pull/66288#discussion_r1452136887

Comment (mswarowsky) [reply modules/mbedtls/CMakeLists.txt]: moved it

File: modules/mbedtls/zephyr_time.c (hunk 2 above)
Comment (mswarowsky) [reply modules/mbedtls/zephyr_time.c]: moved it

**Review by SebastianBoe — DISMISSED** [review]

File: boards/arm/b_u585i_iot02a/doc/index.rst [hunk 3]
```diff
@@ -346,7 +346,7 @@ can be generated using ``b_u585i_iot02a_ns`` as build target.
 
    $ west build -b b_u585i_iot02a_ns path/to/source/directory
 
-Note: When building the ``*_ns`` image with TF-M, ``build/tfm/postbuild.sh`` bash script
+Note: When building the ``*_ns`` image with TF-M, ``build/tfm/api_ns/postbuild.sh`` bash script
 is run automatically in a post-build step to make some required flash layout changes.
 
 Once the build is completed, run the following script to initialize the option bytes.
```
Comment (erwango) [inline boards/arm/b_u585i_iot02a/doc/index.rst:352]: To be fully complete, you'd need to update the pre-flashing instruction as well:
```
-$ ./build/tfm/regression.sh
+$./build/tfm/api_ns/regression.sh
```

**Review by erwango — CHANGES_REQUESTED** [review]
Also, it seems you had a rebase issue with the last "fixup !" commit remaining.
Otherwise LGTM, great work!

File: boards/arm/b_u585i_iot02a/doc/index.rst (hunk 3 above)
Comment (mswarowsky) [reply boards/arm/b_u585i_iot02a/doc/index.rst:352]: good point, changed it

**Review by erwango — DISMISSED** [review]

**Review by SebastianBoe — DISMISSED** [review]

**Review by d3zd3z — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by kartben** [comment]
Could someone please update the documentation? It looks like this was missed as part of this PR.
For example https://docs.zephyrproject.org/latest/services/tfm/overview.html
" Zephyr RTOS has been PSA Certified since Zephyr 2.0.0 with TF-M 1.0, and **is currently integrated with TF-M 1.8.0.**"

@mswarowsky @d3zd3z

**Comment by mswarowsky** [comment]
> Could someone please update the documentation? It looks like this was missed as part of this PR. For example https://docs.zephyrproject.org/latest/services/tfm/overview.html " Zephyr RTOS has been PSA Certified since Zephyr 2.0.0 with TF-M 1.0, and **is currently integrated with TF-M 1.8.0.**"
> 
> @mswarowsky @d3zd3z

Created #67900 I'm not able to add reviewers so pleas @kartben and @d3zd3z feel free to check.

**Pull request merged**